commands.py

```diff
@@ -133,24 +145,13 @@ def on_query_context(self, view, key, *args):
         pipe.view_launched_build = view
         pipe.prepare_copy(window)
 
-        # We cannot call the `hide_panel` directly; it has no effect if we do
-        # so, last I checked in Sublime 2221 and 3103. Not surprising,
-        # considering how most event-loops work. So we employ the (usual) out-
-        # of-band trick.
-        #
-        # It is interesting to note that while `show_panel` accepts the panel
-        # argument, `hide_panel` doesn't. See [1]. Though, [2] gives a hint
-        # that it might work if the panel is specified via contexts instead of
-        # as an argument. But the API (`run_command`) doesn't allow contexts to
-        # be specified, only key bindings in a .sublime-keymap. Hmph.
-        #
         # [1] https://sublimetext.userecho.com/topics/1930-add-panel-param-to-hide_panel-command/
         # [2] https://forum.sublimetext.com/t/output-panel-hotkey/15628/2
-        def hide_panel():
-            window.run_command("hide_panel")
-        sublime.set_timeout(hide_panel, 1)
```

Comment:
Any reason why hiding of the Build panel is dropped?

Reply:
Because you can explicitly disable the build panel with `show_panel_on_build`. We could say that when someone explicitly enables this package with `"buildview.enabled": true`, they should disable the build panel with `"show_panel_on_build": false`.

Due the bug:
1. https://github.com/SublimeTextIssues/Core/issues/679 show_panel_on_build setting isn't loaded from syntax specific settings

Sublime Text is not loading the setting `show_panel_on_build` from the user projects settings. Then, if someone enables this package per project only, they will not be able to stop the build panel from showing up. However, the fix is very simple, as explained on the issue (https://github.com/SublimeTextIssues/Core/issues/679) and can be performed by any user. For example, my Sublime Text is already patched against that.

I got tired of living with those small glitches/bugs which can be extremely easy fixed by the Sublime Text developers. Then, I decided to stop patching our packages with very ugly and hardly working hacks, and let the affected users directly patch Sublime Text files. Of course, only if they can. Otherwise we would still have to keep our packages with those very ugly hacks.